rcl/test/rcl/test_subscription.cpp

```diff
@@ -885,6 +886,431 @@ TEST_F(CLASSNAME(TestSubscriptionFixture, RMW_IMPLEMENTATION), test_bad_return_l
     rcl_subscription_fini(&subscription, this->node_ptr)) << rcl_get_error_string().str;
 }
 
+/* A subscription with a content filtered topic setting.
```

Comment:
mostly filtering test is checked with string type only, i think it is okay for primitive type, but we would want to test with user defined message just in case?

Reply:
The `BasicTypes.msg` is a user-defined message in `test_interface_files`, do you mean we need to test with the `Nested.msg`?

Reply:
Ah, I used the string in the test, I misremembered the BasicTypes is tested in `test_rmw_implementation`.
I will update the following case(`test_subscription_not_initialized_with_content_filtering`) by using the BasicTypes.

Reply:
Updated in 206facde4e4621d2af045e3ff176758f81ad6fda